celery_progress/backend.py

```diff
@@ -52,6 +54,27 @@ def stop_task(self, current, total, exc):
         )
 
 
+class WebSocketProgressRecorder(ProgressRecorder):
+
+    def set_progress(self, current, total):
+        super().set_progress(current, total)
+        channel_layer = get_channel_layer()
+        task_id = self.task.request.id
+        async_to_sync(channel_layer.group_send)(
+            task_id,
+            {'type': 'update_task_progress', 'data': {**Progress(task_id).get_info()}}
+        )
+
+    def stop_task(self, current, total, exc):
+        super().stop_task(current, total, exc)
+        channel_layer = get_channel_layer()
+        task_id = self.task.request.id
+        async_to_sync(channel_layer.group_send)(
+            task_id,
+            {'type': 'update_task_progress', 'data': {**Progress(task_id).get_info()}}
+        )
```

Comment:
nit: looks like this most of this method could be shared with the previous one

Reply:
I agree, I can definitely slim this down to a function and call it from within the two methods.